alphagov/paas-rds-metric-collector
# [#158333555] Stop double emitting metrics

## What

We want to only emit metrics from a single instance of this collector. We have done this using the Locket client and server, as done by Cloud Foundry components such as TPS.

## How to review

* code review
* run `make unit`
* check integration tests passed:
  https://concourse.build.ci.cloudpipeline.digital/teams/main/pipelines/rds-metric-collector
  Note: the integration tests have been flakey. A separate story has been raised [#159328199](https://www.pivotaltracker.com/story/show/159328199).

Optionally, you can deploy it with https://github.com/alphagov/paas-cf/pull/1462 and go poke at the logs.

## Who

Anyone but me or @richardTowers
